test/trip_dispatch_test.rb

```diff
@@ -122,4 +128,103 @@ def build_test_dispatcher
       end
     end
   end
-end
+
+  describe "Requesting a Trip" do
+    before do
+      @dispatcher = build_test_dispatcher
+      @requested_trip = @dispatcher.request_trip(8)
+    end
+
+    it "creates a new Trip instance" do
+      expect(@requested_trip).must_be_kind_of RideShare::Trip
+    end
+
+    it "checks that end_time, cost, and rating are set to nil at initialization" do
+      expect(@requested_trip.end_time).must_equal nil
+      expect(@requested_trip.cost).must_equal nil
+      expect(@requested_trip.rating).must_equal nil
+    end
+
+    it "raises an ArgumentError if passenger_id is not valid" do
+      expect{@dispatcher.request_trip(0)}.must_raise ArgumentError
+      expect{@dispatcher.request_trip(-2)}.must_raise ArgumentError
+    end
+
+    # Testing provided data (where most eligible driver had no trips)
+    it "correctly finds first available driver when a driver has no trips or trips is nil" do
+      expect(@requested_trip.driver.name).must_equal "Driver 3 (no trips)"
+    end
+
+    it "sets the selected driver's status to :UNAVAILABLE" do
+      expect(@requested_trip.driver.status).must_equal :UNAVAILABLE
+    end
+  end
+
+  describe "Intelligent Dispatching Methods" do
+    # Using new directory and data
+    TEST_DATA_DIRECTORY2 = 'test_2/test_data'
+    def build_test_dispatcher2
+      return RideShare::TripDispatcher.new(
+        directory: TEST_DATA_DIRECTORY2
+      )
+    end
+
+    describe "available_drivers method" do
+      before do
+        @dispatcher2 = build_test_dispatcher2
+        @requested_trip = @dispatcher2.request_trip(4)
+      end
+
+      it "returns an array when calling available_drivers" do
+        expect(@dispatcher2.available_drivers).must_be_kind_of Array
+      end
+
+      it "returns the correct number of available drivers" do 
+        previous_available_drivers = @dispatcher2.available_drivers
+        expect(previous_available_drivers.length).must_equal 6
+
+        # Make sure that pool of available drivers decrements when a driver is requested  
+        @requested_trip2 = @dispatcher2.request_trip(5)
+
+        current_available_drivers = @dispatcher2.available_drivers
+        expect(current_available_drivers.length).must_equal 5
+      end
+
+      it "confirms that every driver in returned available drivers list has :AVAILABLE status" do
+        @requested_trip2 = @dispatcher2.request_trip(8)
+        def check_all_statuses 
+          @dispatcher2.available_drivers.each do |driver|
+            if driver.status == :UNAVAILABLE
+              return false
+            else 
+              return true
+            end
+          end
```

Comment:
This method doesn't actually check all statuses.  It only checks the first status since it `return`s on the first time through the `each` loop.  

You probably wanted to use `.all?` instead of `.each`:
```suggestion
          @dispatcher2.available_drivers.all? do 
            driver.status != :UNAVAILABLE
          end
```